bionic/provider.py

```diff
@@ -31,16 +31,23 @@ def __init__(
             self, names,
             protocols=None, code_version=None, orig_flow_name=None,
             should_persist=None, should_memoize=None, is_default_value=None,
-            docs=None):
+            docs=None, is_volatile=None):
         self.names = names
         self.protocols = protocols
         self.docs = docs
         self.code_version = code_version
         self.orig_flow_name = orig_flow_name
-        self.should_persist = should_persist
-        self.should_memoize = should_memoize
+        self._should_persist = should_persist
+        self._should_memoize = should_memoize
+        self.is_volatile = is_volatile
         self.is_default_value = is_default_value
 
+    def should_persist(self):
```

Comment:
It seems confusing to have a method with the same name as an attribute, but a subtly different meaning.

Also, if we disable persistence for volatile entities, that means we won't do the save-to-disk-and-reload step after computing them, right? I'm not sure if that's what we want.

Reply:
I thought I changed the variable name to `_can_persist` and `_can_memoize`, I'll make that change. 

We don't have to disable persistence for volatile entities, it's just that we won't ever use the cached value. I thought the entire purpose of save-to-disk-and-reload was to have consistency between different runs (the first time it is computed vs the next time it is reloaded from the disk). Or are you suggesting there might be some implications down the dependencies, something like if they cache a part of the value and also use it directly from the entity?

Reply:
My concern here was that the save-reload process will subtly change the value, which means that adding `@volatile` to an entity will change that entity's value. It's not clear to me if that's desirable/acceptable.

But also, even if an entity is volatile, we still want to be able to hash its output value, right? So we still need to write it to disk anyway. But maybe we should be writing it to a temporary location.

(Sorry if I'm missing something, I'm running late and writing this in a hurry! I'll get to the other outstanding comments tomorrow.)

Reply:
We don't need to hash the output value yet since we won't use it anywhere. If or when we introduce something `@digestible`, it would change though and maybe we should reconsider the whole hashing on persist thing.

Reply:
Talked offline, here is the discussion result for posterity. 

It makes sense for volatile to persist with `@persist(True)` decorator since even if we don't use the persisted value, the children can still use the hash and not recompute if not needed. Although it would be nice to not persist the value, we can make this change at a later date, perhaps when we introduce `@digestable` where we write the output to a tmp directory instead of the caching directory. 

Also, going to change the names of the variables to `_can_memoize` and `_can_persist`.